pkg/endpoint/log.go

```diff
@@ -115,3 +130,61 @@ func (e *Endpoint) UpdateLogger(fields map[string]interface{}) {
 
 	atomic.StorePointer(&e.logger, unsafe.Pointer(l))
 }
+
+func (e *Endpoint) updatePolicyLogger(fields map[string]interface{}) {
+	pv := atomic.LoadPointer(&e.policyLogger)
+	policyLogger := (*logrus.Entry)(pv)
+	// e.Options check needed for unit testing.
+	if policyLogger == nil && e.Options != nil && e.Options.IsEnabled(option.DebugPolicy) {
+		policyLogOnce.Do(func() {
+			maxSize := 10 // 10 MB
+			if ms := os.Getenv("CILIUM_DBG_POLICY_LOG_MAX_SIZE"); ms != "" {
+				if ms, err := strconv.Atoi(ms); err == nil {
+					maxSize = ms
+				}
+			}
+			maxBackups := 3
+			if mb := os.Getenv("CILIUM_DBG_POLICY_LOG_MAX_BACKUPS"); mb != "" {
+				if mb, err := strconv.Atoi(mb); err == nil {
+					maxBackups = mb
+				}
+			}
+			lumberjackLogger := &lumberjack.Logger{
+				Filename:   filepath.Join(option.Config.StateDir, "endpoint-policy.log"),
+				MaxSize:    maxSize,
+				MaxBackups: maxBackups,
+				MaxAge:     28, // days
+				LocalTime:  true,
+				Compress:   true,
+			}
+			policyLog.SetOutput(lumberjackLogger)
+			policyLog.SetLevel(logrus.DebugLevel)
+		})
+		policyLogger = logrus.NewEntry(policyLog)
+	}
+	if policyLogger == nil || e.Options == nil {
+		return
+	}
+
+	if !e.Options.IsEnabled(option.DebugPolicy) {
+		policyLogger = nil
```

Comment:
Can we fold the `e.Options.IsEnabled` check with L 165, and move all the checks earlier at L137?

Reply:
I don't think we can, as `policyLogger` can be set on the block right above the line 165, so the nil check can not be done before that. As for the lines 169-170, these require the `atomic.StorePointer()` at the end of the function.